### ISSUE TITLE
Menus opened via xfce4-popup-commands don't get keyborad focus on opening

### ISSUE THREAD

**Issue by Tamaranch** [labwc/labwc#2467]
### Problem description

When running e.g. xfce4-popup-applicationsmenu (shipped with xfce4-panel), the xfce app menus opens, but the menu doesn't get focus. It does get focus with wayfire 0.9.0.

Keyboard focus is set to on-demand in xfce4-panel.

### Steps to reproduce

1. Rune xfce4-panel
2. Add xfce4-applicationsmenu-plugin
3. Run xfce4-popup-applicationsmenu
4. Try to navigate in the menu via the arrow keys: it does not work, because the menu doesn't get focus.

### labwc build source

Release

### labwc version

0.8.2 or 0.8.2-29-g1550eb9

### labwc environment

From a TTY or some display manager like lightdm

### Distribution

Arch Linux

**Comment by johanmalm** [comment]
Similar to https://github.com/labwc/labwc/issues/1572

I have tried playing around with this before but:
1. I couldn't find a way of grabbing the keyboard when popups open. Welcome help with this.
2. IIRC the protocol states the child surface should inherit the interactivity of their parent. I would personally be happy to consider popups special in this case and only apply that rule to sub-surfaces.

**Comment by Consolatis** [comment]
There is a [xdg_popup.grab()](https://gitlab.freedesktop.org/wayland/wayland-protocols/-/blob/main/stable/xdg-shell/xdg-shell.xml?ref_type=heads#L1253) request which might be related or not, not sure about a layershell parent + xdg_popup combination.

**Comment by johanmalm** [comment]
> There is a [xdg_popup.grab()](https://gitlab.freedesktop.org/wayland/wayland-protocols/-/blob/main/stable/xdg-shell/xdg-shell.xml?ref_type=heads#L1253) request which might be related or not, not sure about a layershell parent + xdg_popup combination.

Yes. I played with that. IIRC it caused focus problems on some panels’ tooltips.


> The current workaround approach seems to be to request `EXCLUSIVE`, open the popup, return back to `ON_DEMAND`.

Yes

**Comment by tokyo4j** [comment]
@vyivel Does wlroots provide a way to listen for `xdg_popup.grab` requests? I found `wlr_seat.events.keyboard_grab_begin`, but I couldn't manage to tell whether the event comes from `xdg_popup.grab` requests or DnD.

**Comment by vyivel** [comment]
wlroots doesn't provide a way to handle `xdg_popup.grab` requests currently, no. It will be added (and the original issue fixed as well) with https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/4950.

**Comment by johanmalm** [comment]
Thanks @vyivel 😄

**Comment by gtellalov** [comment]
I have a problem with window focus as well. It could be related. 
I have a keyboard shortuct in the config file       <action name="Execute" command="xfce4-appfinder" />
and when I press the key the appfinder appears but it's not focused.
Probably also related - sometimes when switching windows the new window doesn't get focus as well, even though it appears in the foreground.

**Comment by tokyo4j** [comment]
@gtellalov Would you mind creating another issue for it? I think yours is another different topic as the issue discussed here is about how we handle `xdg_popup`, but not toplevels (windows).

Also, comparing labwc versions between the last git commit and 0.8.2 will be helpful as #2455 made a big change in focus management. I tried adding the `xfce4-appfinder` shortcut, but I couldn't reproduce the issue.

**Comment by gtellalov** [comment]
Created as #2534
